src/Illuminate/Auth/AuthServiceProvider.php

```diff
@@ -46,7 +47,12 @@ protected function registerAuthenticator()
      */
     protected function registerUserResolver()
     {
-        $this->app->bind(AuthenticatableContract::class, fn ($app) => call_user_func($app['auth']->userResolver()));
+        $this->app->bind(AuthenticatableContract::class, function ($app) {
+            $user = call_user_func($app['auth']->userResolver());
+            throw_if(! $user, new UnauthorizedException);
```

Comment:
Possible breaking change. This means any custom code resolving the `Authenticatable` contract now will now throw `UnauthorizedException` instead of `null`.

Reply:
Yea, I thought about not even including this in the PR but I personally feel that being able to pull the authenticated user out of the container is a huge productivity boost. Still, I realize it’s a bigger deal so I’m happy to split that out to a separate PR if the `->userOrFail()` method is an easier change to merge.

Reply:
What’s interesting is that you can actually use the Authenticatable type hint today without this change. Type hint an Authenticatable variable in a controller and you’ll get the logged in user. The only issue with the current functionality is the `TypeError` when not logged in.

Reply:
Can you PR ensure the following still green?

```
<?php

namespace Illuminate\Tests\Integration\Auth;

use Illuminate\Contracts\Auth\Authenticatable;
use Orchestra\Testbench\Factories\UserFactory;
use Orchestra\Testbench\TestCase;

class AuthenticatableTest extends TestCase
{
    /**
     * Define database migrations.
     *
     * @return void
     */
    protected function defineDatabaseMigrations()
    {
        $this->loadLaravelMigrations();
    }

    public function testItCanReturnNullWhenUserIsNotAuthenticated()
    {
        $this->assertNull($this->app->make(Authenticatable::class));
    }

    public function testItCanReturnUserWhenUserIsNotAuthenticated()
    {
        $user = UserFactory::new()->create();

        $this->app->make('auth')->resolveUsersUsing(fn () => $user);

        $this->assertTrue($user->is($this->app->make(Authenticatable::class)));
    }
}
```

![image](https://github.com/laravel/framework/assets/172966/5ee875e3-f92c-4bcb-8f60-a2f9cf73f31e)